### PR TITLE
chore(serde): adding borsh for witness serde

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**

### CODE REVIEW

File: core/Cargo.toml [hunk 1]
```diff
@@ -15,10 +15,12 @@ bitvec = { version = "1", default-features = false, features = ["alloc"] }
 hex = { version = "0.4.3", default-features = false, features = ["alloc"] }
 ruint = { version = "1.12.1", default-features = false }
 arrayvec = { version = "0.7", default-features = false }
+borsh = { version = "1.5.3", default-features = false, features = ["derive"], optional = true}
```
Comment (rphmeier) [inline core/Cargo.toml:18]:
```suggestion
borsh = { version = "1.5.3", default-features = false, features = ["derive"], optional = true }
```

**Review by rphmeier — COMMENTED** [review]
Looks good (after you address the nits)

Thanks for this PR! Though note that the API here isn't stable, so it may change. We will preserve the `borsh` serialization.

File: nomt/Cargo.toml [hunk 2]
```diff
@@ -30,6 +30,7 @@ libc = "0.2.155"
 criterion = { version = "0.3", optional = true }
 thread_local = "1.1.8"
 cfg-if = "1.0.0"
+borsh = { version = "1.5.3", default-features = false, features = ["derive"], optional = true}
```
Comment (rphmeier) [inline nomt/Cargo.toml:33]:
```suggestion
borsh = { version = "1.5.3", default-features = false, features = ["derive"], optional = true }
```

**Comment by dzmitry-lahoda** [comment]
Fixing 
```
 Updating crates.io index
error: failed to select a version for `syn`.
    ... required by package `borsh-derive v1.5.3`
    ... which satisfies dependency `borsh-derive = "~1.5.3"` of package `borsh v1.5.3`
    ... which satisfies dependency `borsh = "^1.5.3"` of package `nomt-core v0.1.0 (/home/runner/work/nomt/nomt/core)`
    ... which satisfies path dependency `nomt-core` (locked to 0.1.0) of package `nomt v0.1.0 (/home/runner/work/nomt/nomt/nomt)`
    ... which satisfies path dependency `nomt` (locked to 0.1.0) of package `benchtop v0.1.0 (/home/runner/work/nomt/nomt/benchtop)`
versions that meet the requirements `^2.0.81` are: 2.0.90, 2.0.89, 2.0.88, 2.0.87, 2.0.86, 2.0.85, 2.0.84, 2.0.83, 2.0.82, 2.0.81

all possible versions conflict with previously selected packages.

  previously selected package `syn v2.0.59`
    ... which satisfies dependency `syn = "^2.0.8"` (locked to 2.0.59) of package `clap_derive v4.5.4`
    ... which satisfies dependency `clap_derive = "=4.5.4"` (locked to 4.5.4) of package `clap v4.5.4`
    ... which satisfies dependency `clap = "^4.4.8"` (locked to 4.5.4) of package `benchtop v0.1.0 (/home/runner/work/nomt/nomt/benchtop)`

failed to select a version for `syn` which could resolve this conflict
Error: Process completed with exit code 101.
```

**Comment by rphmeier** [comment]
I think if you commit the latest `Cargo.lock` files this should turn green and we can merge it.

**Comment by dzmitry-lahoda** [comment]
Borsh 1.5.2 was broken released (the forget to update syn). Borsh 1.5.3 updated syn, but it incompatible with benchtop for now (because it builds some "old" codebases). I tried to force borsh 1.5.1 version, but unfortunately borsh has flexible dependency onto borsh-derive, so it picks 1.5.2 derive create, which is broken.

So I used borsh 1.4.0 release.

**Review by pepyakin — APPROVED** [review]

**Pull request merged**